lib/routes/idolmaster/news.ts

```diff
@@ -0,0 +1,111 @@
+    },
+    description: `**Brand**
+    | THE IDOLM@STER | シンデレラガールズ | ミリオンライブ！ | SideM | シャイニーカラーズ | 学園アイドルマスター | その他 |
+    | -------------- | --------------- | ------------- | ----- | --------------- | ----------------- | ----- |
+    | IDOLMASTER | CINDERELLAGIRLS | MILLIONLIVE | SIDEM | SHINYCOLORS | OTHER |
+
+    **Category**
+    | ゲーム | ライブ・イベント | アニメ | 配信番組 | ラジオ | グッズ | コラボ・キャンペーン | ミュージック | ブック・コミック | メディア | その他 |
+    | ----- | ------------- | ----- | ------- | ----- | ----- | ----------------- | --------- | -------------- | ------ | ----- |
+    | GAME | LIVE-EVENT | ANIME | LIVESTREAM | RADIO | GOODS | COLLABO-CAMP | CD | BOOK | MEDIA | OTHER |
+    `,
+    features: {
+        requireConfig: false,
+        requirePuppeteer: false,
+        antiCrawler: false,
+        supportBT: false,
+        supportPodcast: false,
+        supportScihub: false,
+    },
+    radar: [
+        {
+            source: ['idolmaster-official.jp/news'],
+            target: '/news',
+        },
+    ],
+    name: 'ニュース News',
+    maintainers: ['keocheung'],
+    handler,
+};
+
+const apiUrl = 'https://cmsapi-frontend.idolmaster-official.jp';
+
+async function handler(ctx: Context): Promise<Data> {
+    const tokenUrl = `${apiUrl}/sitern/api/cmsbase/Token/get`;
+    const tokenRsp = await got(tokenUrl);
+    const token = tokenRsp.data.data.token;
+
+    const options: {
+        category: string[];
+        subcategory?: string | string[];
+        brand?: string | string[];
+    } = {
+        category: ['NEWS'],
+    };
+
+    const routeParams = ctx.req.param('routeParams');
+    if (routeParams) {
+        const routeParams = querystring.parse(ctx.req.param('routeParams'));
+        options.subcategory = toUpperCase(routeParams.category);
+        options.brand = toUpperCase(routeParams.brand);
+    }
+
+    const limitParam = ctx.req.query('limit');
+    let limit = limitParam ? Number.parseInt(limitParam) : 12;
+    if (limit > 30) {
+        limit = 30;
+    }
+    const listUrl = `${apiUrl}/sitern/api/idolmaster/Article/list?site=jp&ip=idolmaster&token=${token}&sort=desc&data=${JSON.stringify(options)}&limit=${limit}&start=0`;
+    const listnRsp = await got(listUrl);
+    const articleList = listnRsp.data.data.article_list;
+
+    let items = articleList.map(
+        (article): DataItem => ({
+            title: article.title,
+            link: article.url,
+            pubDate: timezone(parseDate(article.dspdate), +9),
+            category: article.categories.subcategory.map((cat) => cat.name),
+        })
+    );
+
+    items = await Promise.all(
+        items.map((item: DataItem) =>
+            cache.tryGet(item.link, async () => {
+                const rsp = await got(item.link);
+                const content = load(rsp.data);
+                item.description = `<div lang="ja">${content('div[class^="style_inner_content"]').html()?.replaceAll('<img src="', `<img src="${apiUrl}/sitern/api/idolmaster/Image/get?path=`)}</div>`;
```

Comment:
Please use data from the embedded JSON instead.
```suggestion
                const content = load(rsp.data);
                const nextData = JSON.parse(content('script#__NEXT_DATA__').text());
                item.description = `<div lang="ja">${nextData.props.pageProps.data.content?.replaceAll('<img src="', `<img src="${apiUrl}/sitern/api/idolmaster/Image/get?path=`)}</div>`;
```